pool/pool.go

```diff
@@ -1302,7 +1313,7 @@ func (p *Pool) openDefaultSession(ctx *callContext) error {
 	tok, ok := p.cache.Get(cacheKey)
 	if !ok {
 		// init new session
-		err := initSessionForDuration(ctx, &tok, ctx.client, p.stokenDuration)
+		err := initSessionForDuration(ctx, &tok, ctx.client, p.stokenDuration, ctx.key)
```

Comment:
So if pool invoker didn't set a key, then `ctx.key` is `nil`, then default key is used. Otherwise use key from the context, right?

Reply:
Here `ctx.key` is already set default key or provided one